tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -808,12 +807,57 @@ TEST_CASE("Test Tap position optimizer") {
 
             //     u_set = 0.4;
             //     u_set = 0.01;
-            //     check_b = check_exact_per_strategy(3, 4, 3);
+            //     check_b = test::check_exact_per_strategy(3, 4, 3);
             // }
 
             regulator_b.update({.id = 4, .u_set = u_set, .u_band = u_band});
         }
 
+        SUBCASE("line drop compensation") {
+            state_b.rank = 0;
+            state_b.u_pu = [&state_b, &regulator_b](ControlSide side) {
+                CHECK(side == regulator_b.control_side());
+
+                // tap pos closer to tap_max at tap side <=> lower voltage at control side
+                return static_cast<DoubleComplex>(
+                    test::normalized_lerp(state_b.tap_pos, state_b.tap_max, state_b.tap_min));
+            };
+            state_b.i_pu = [&state_b, &regulator_b](ControlSide side) {
+                CHECK(side == regulator_b.control_side());
+                auto const value = test::normalized_lerp(state_b.tap_pos, state_b.tap_min, state_b.tap_max);
+                return DoubleComplex{value, value};
+            };
+
+            double const u_set = 0.5;
+            double const u_band = 0.76;
+
+            state_b.tap_min = 1;
+            state_b.tap_max = 5;
+            state_b.tap_pos = 3;
+            double line_drop_compensation_r = 0.0;
+            double line_drop_compensation_x = 0.0;
+
+            SUBCASE("no line drop compensation") { check_b = test::check_exact_per_strategy(3, 4, 2); }
+            SUBCASE("resistance") {
+                line_drop_compensation_r = 0.5;
+                check_b = test::check_exact_per_strategy(3, 5, 3);
+            }
+            SUBCASE("positive reactance") {
+                line_drop_compensation_x = 0.125;
+                check_b = test::check_exact_per_strategy(3, 5, 2);
+            }
+            SUBCASE("negative reactance") {
+                line_drop_compensation_x = -0.5;
+                check_b = test::check_exact_per_strategy(3, 5, 3);
+            }
```

Comment:
i could also add tests like what happens if i switch off `i_pu` or what happens if i make `i_pu` a purely real or purely imaginary function. thoughts?

Reply:
follow-up if necessary in #595